CONTRIBUTING.adoc

```diff
@@ -196,6 +196,117 @@ Follow these steps to ensure that virtual box interfaces are unmanaged:
 
     $ sudo nmcli con reload
 
+=== Develop and test using a docker-in-docker cluster
+
+It's possible to run an OpenShift multinode cluster on a single host
+via docker-in-docker (dind).  Cluster creation is cheaper since each
+node is a container instead of a VM.  This was implemented primarily
+to support multinode network testing, but may prove useful for other
+use cases.
+
+To run a dind cluster in a VM, follow steps 1-3 of the Vagrant
+instructions and then execute the following:
+
+        $ export OPENSHIFT_DIND_DEV_CLUSTER=true
+        $ vagrant up
+
+Bringing up the VM for the first time will take a while due to the
+overhead of package installation, building docker images, and building
+openshift.  Assuming the 'vagrant up' command completes without error,
+a dind OpenShift cluster should now be running on the VM.  To access
+the cluster, login to the VM:
+
+        $ vagrant ssh
+
+Once on the VM, the 'oc' and 'openshift' commands can be used to
+interact with the cluster:
+
+        $ oc get nodes
```

Comment:
FWIW, most oc commands seem to work, but "oc project foo" gives an error. ("oc new-project foo" creates the project successfully, but then gives an error when trying to switch to it):

    [vagrant@openshift-dind-host vagrant]$ oc new-project foo
    error: open /tmp/openshift-dind-cluster/openshift/openshift.local.config/master/admin.kubeconfig: permission denied